onboarding/source/amb_light_service.c

```diff
@@ -52,20 +56,22 @@ uint8_t initLightService(void) {
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
-    adcData_t adc_data;
-    adcData_t *adc_data_ptr = &adc_data;
+    adcData_t adcData;
+    light_event_t *eventReceived = NULL;
```

Comment:
`xQueueReceive` will copy the event to the address pointed to by `eventReceived`. So `eventReceived` cannot be NULL. It should point to a valid block of memory that you've allocated. I'd suggest just declaring a `light_event_t` variable like `light_event_t event` and passing `&event` into `xQueueReceive`.